From 35ec6c7689fabf79260e5966fd6a3d57811a8611 Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Fri, 7 Oct 2016 16:45:29 -0700
Subject: [PATCH] DOCKER-946 docker pull authentication errors are no longer
 reported Reviewed by: Todd Whiteman <todd.whiteman@joyent.com> Approved by:
 Todd Whiteman <todd.whiteman@joyent.com>

---
 lib/backends/sdc/images.js | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/lib/backends/sdc/images.js b/lib/backends/sdc/images.js
index b266573..54ae62d 100644
--- a/lib/backends/sdc/images.js
+++ b/lib/backends/sdc/images.js
@@ -1415,13 +1415,14 @@ function pullImage(opts, callback) {
                 status: format('Error pulling image: (%s) %s host not found',
                     err.code, opts.rat.index.name)
             };
-        } else if (err.code === 'UnauthorizedError') {
+        } else if (err.code === 'UnauthorizedError'
+                || err.code === 'Unauthorized') {
             payload = {
                 status: format('Unauthorized error from registry %s trying to '
                     + 'pull %s (%s)', opts.rat.index.name, opts.rat.remoteName,
                     opts.req_id)
             };
-        } else if (err.code === 'NotFoundError') {
+        } else if (err.code === 'NotFoundError' || err.code === 'NotFound') {
             // E.g. `docker pull quay.io/no-such-user`.
             payload = {
                 status: format('Not Found (404) error from registry %s trying '
-- 
2.21.0

